src/wallet/wallet.cpp

```diff
@@ -2544,9 +2544,10 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend,
 
                         // Reserve a new key pair from key pool
                         CPubKey vchPubKey;
-                        bool ret;
-                        ret = reservekey.GetReservedKey(vchPubKey, true);
-                        assert(ret); // should never fail, as we just unlocked
+                        if (!reservekey.GetReservedKey(vchPubKey, true)) {
+                            strFailReason = _("Can't generate a change-address key. Please call keypoolrefill first.");
```

Comment:
As a note. This could only fail with this error if the wallet is locked and the keypool is empty. `GetReservedKey` fills the pool if it's needed.

Reply:
Of course. 
As written in the PR description, that was the reason to include bitcoin#17219: make the function callable even with locked/encrypted wallet (and return an error only if the keypool is empty and a change address/script is required).

Reply:
yeah. It was more a note for my future self. Wasn't expecting it in this first iteration of `fundrawtransaction`, the change was merged in upstream pretty recently.